service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala

```diff
@@ -146,28 +160,36 @@ class PrestoQueryExecutoryFactory extends QueryExecutoryFactory {
     |}
   """.stripMargin
 
-  override def fromJson(configJson: JValue): MahaServiceConfig.MahaConfigResult[QueryExecutor] =  {
+  override def fromJson(configJson: JValue)(implicit context:MahaServiceConfigContext): MahaServiceConfig.MahaConfigResult[QueryExecutor] =  {
     import org.json4s.scalaz.JsonScalaz._
 
-    val dataSourceFactoryClassResult: MahaServiceConfig.MahaConfigResult[String] = fieldExtended[String]("dataSourceFactoryClass")(configJson)
-    val dataSourceFactoryConfigResult: MahaServiceConfig.MahaConfigResult[JValue] = fieldExtended[JValue]("dataSourceFactoryConfig")(configJson)
+    val dataSourceNameResult: MahaServiceConfig.MahaConfigResult[String] = fieldExtended[String]("dataSourceName")(configJson).map(_.toLowerCase)
     val jdbcConnectionFetchSizeOptionResult: MahaServiceConfig.MahaConfigResult[Option[Int]] = fieldExtended[Option[Int]]("jdbcConnectionFetchSize")(configJson)
     val lifecycleListenerFactoryClassResult: MahaServiceConfig.MahaConfigResult[String] = fieldExtended[String]("lifecycleListenerFactoryClass")(configJson)
     val lifecycleListenerFactoryConfigResult: MahaServiceConfig.MahaConfigResult[JValue] = fieldExtended[JValue]("lifecycleListenerFactoryConfig")(configJson)
     val prestoQueryTemplateFactoryNameResult: MahaServiceConfig.MahaConfigResult[String] = fieldExtended[String]("prestoQueryTemplateFactoryName")(configJson)
     val prestoQueryTemplateFactoryConfigResult: MahaServiceConfig.MahaConfigResult[JValue] = fieldExtended[JValue]("prestoQueryTemplateFactoryConfig")(configJson)
 
+    import _root_.scalaz._
+    import syntax.validation._
+
+    for {
+      dataSourceName <- dataSourceNameResult
+    } yield  {
+      if(!context.dataSourceMap.contains(dataSourceName)) {
+        return Failure(List(ServiceConfigurationError(s"Failed to find presto dataSourceName $dataSourceName in dataSourceMap"))).toValidationNel.asInstanceOf[MahaConfigResult[QueryExecutor]]
+      }
+    }
+
     val jdbcConnetionResult : MahaServiceConfig.MahaConfigResult[JdbcConnection] = for {
-      dataSourceFactoryClass <- dataSourceFactoryClassResult
-      dataSourceFactoryConfig <- dataSourceFactoryConfigResult
-      dataSourceFactory <-  getFactory[DataSourceFactory](dataSourceFactoryClass, this.closer)
-      dataSource <- dataSourceFactory.fromJson(dataSourceFactoryConfig)
+      dataSourceName <- dataSourceNameResult
+      dataSource <- context.dataSourceMap.get(dataSourceName).successNel[Option[DataSource]].asInstanceOf[MahaServiceConfig.MahaConfigResult[Option[DataSource]]]
       jdbcConnectionFetchSizeOption <- jdbcConnectionFetchSizeOptionResult
     } yield {
       if(jdbcConnectionFetchSizeOption.isDefined) {
-        new JdbcConnection(dataSource, jdbcConnectionFetchSizeOption.get)
+        new JdbcConnection(dataSource.get, jdbcConnectionFetchSizeOption.get)
```

Comment:
This is unsafe get on dataSource.  Check if it is defined first.

Reply:
datasource.get is already validated above if not defined, execution return the MahaServiceError list.

Reply:
Can you add a test for this?

Reply:
Added tests for checking unknown datasource